File: session/02_double_dispatch/java?/Scissor.java
public class Scissor implements Beatable {
	@Override
	public String beats(Beatable other) {
		if (other.getClass() == Rock.class) {
			return "scissor loses";
		}

		if (other.getClass() == Paper.class) {
			return "scissor wins";
		}
		return "draw";
	}
}

File: 02_double_dispatch/c#/02_after_pattern_matching/Scissor.cs
public class Scissor : Beatable {
	public string Beats(Beatable other) {
    switch (other) {
      case Rock r:
			  return "scissor loses";
      case Paper p:
			  return "scissor wins";
      default:
		    return "draw";
    }
	}
}
